autoscheduler/scraper/tests/api_tests.py

```diff
@@ -265,3 +269,33 @@ def test_api_course_search_gives_correct_results_c(self):
         # Assert
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_3(self):
+        """ Tests that /api/course/search?search=CSCE%203&term=201731 gives correct
+            output
+        """
+        # Arrange
+        expected = {'results': ['CSCE 310', 'CSCE 315']}
+        data = {'search': 'CSCE%203', 'term': '201731'}
```

Comment:
Does this actually test that the 3 is processed? I think it would be better to add another CSCE course in the 100s or 200s that would not be matched by this test

Reply:
Done, I also added a test to make sure a lowercase search with numbers (such as "csce 3") is handled correctly